cmake/Modules/FindAzureSDK_EP.cmake

```diff
@@ -83,38 +85,86 @@ if (NOT AZURESDK_FOUND)
     endif()
 
     if (WIN32)
-      set(CFLAGS_DEF "${CMAKE_C_FLAGS}")
-      set(CXXFLAGS_DEF "${CMAKE_CXX_FLAGS}")
+      set(CFLAGS_DEF " /EHsc -I${TILEDB_EP_INSTALL_PREFIX}/include /Dazure_storage_lite_EXPORTS /DCURL_STATICLIB=1 /DWIN32 ${CMAKE_C_FLAGS}")
```

Comment:
- Needs to be `if (MSVC)`
- Remove `/DWIN32`, cmake already adds that by default
- What is the reason to set `/EHsc` here? CMake sets it by default, and ideally we should leave exception mode alone in case client libraries override in their build system (if you absolutely need to override for some reason, please add a comment explaining why, and use `target_compile_options` rather than setting CFLAGS)